### PR TITLE
Fixed issues with "responsive" option, added tableBodyHeight and tableBodyMaxHeight options, added new responsive mode "simple"

### PR DESCRIPTION
**Pull request by patorjk**
This PR aims to fix the issues with the "responsive" option. These are:

* There is no easy way to set the height or maxHeight of the table.
* The "stacked" mode appears misaligned since version 2.14.0.
* The "stacked" mode doesn't handle long strings of text (it overflows off the table).
* The responsive option names are confusing.

This PR splits the setting of the table height away from the responsive parameter. It creates 3 new responsive options:

* "vertical": This mode replaces "stacked" mode. Long strings of text are now correctly handled and alignment appears correct.
* "standard": This mode replaces "scrollMaxHeight" mode.
* "simple": This mode mimics the design in this issue: https://github.com/gregnb/mui-datatables/issues/1188

To deal with table height, two new options are introduced:

* tableBodyHeight: Set the height of the table.
* tableBodyMaxHeight: Set the max height of the table.

If both are set, the table will use tableBodyMaxHeight (though maybe it makes more sense to use tableBodyHeight?). These options are subtly different. I've updated the "Simple" example so that it's easy to toggle between various modes and set various heights.

All old responsive modes will continue to be supported in v3, but will be deprecated.

These updates can be tried [live in this codesandbox](https://codesandbox.io/s/github/gregnb/mui-datatables/tree/responsiveRefactor) - See the "Simple" example. Feedback is greatly appreciated.

### CODE REVIEW

File: src/components/TableBody.js [hunk 1]
```diff
@@ -8,13 +8,28 @@ import TableSelectCell from './TableSelectCell';
...
+import classNames from 'classnames';
 
-const defaultBodyStyles = {
+const defaultBodyStyles = theme => ({
   root: {},
   emptyTitle: {
     textAlign: 'center',
   },
-};
+  lastStackedCell: {
```
Comment (wdh2100) [inline src/components/TableBody.js:18]: great 👏 👏 👏

File: src/components/TableBody.js [hunk 2]
```diff
@@ -221,15 +236,21 @@ class TableBody extends React.Component {
...
             return (
               <React.Fragment key={rowIndex}>
                 <TableBodyRow
-                  {...(options.setRowProps ? options.setRowProps(row, dataIndex, rowIndex) : {})}
+                  {...bodyClasses}
                   options={options}
                   rowSelected={isRowSelected}
                   isRowSelectable={isRowSelectable}
                   onClick={this.handleRowClick.bind(null, row, { rowIndex, dataIndex })}
+                  className={classNames({
```
Comment (wdh2100) [inline src/components/TableBody.js:249]: What happens to the deprecated options?

File: src/MUIDataTable.js [hunk 3]
```diff
@@ -369,18 +378,11 @@ class MUIDataTable extends React.Component {
       );
       this.options.selectableRows = this.options.selectableRows ? 'multiple' : 'none';
     }
-    if (
-      ['scrollMaxHeight', 'scrollFullHeight', 'stacked', 'stackedFullWidth', 'scrollFullHeightFullWidth'].indexOf(
-        this.options.responsive,
-      ) === -1
-    ) {
+    if (['standard', 'vertical', 'simple'].indexOf(this.options.responsive) === -1) {
```
[the remark this answers is not in the document]
Comment (patorjk) [reply src/MUIDataTable.js:381]: I agree, in my next push I'll update this.

File: src/MUIDataTable.js [hunk 4]
```diff
@@ -1500,11 +1497,21 @@ class MUIDataTable extends React.Component {
...
+    let maxHeight = this.options.tableBodyMaxHeight;
     let responsiveClass;
 
+    if (this.options.tableBodyMaxHeight && this.options.tableBodyHeight){
+      warnInfo('Both tableBodyMaxHeight and tableBodyHeight, tableBodyMaxHeight will be used.');
+    }
+
     switch (responsiveOption) {
-      // DEPRECATED: This options is beign transitioned to `responsiveScrollMaxHeight`
+      case 'standard':
```
[the remark this answers is not in the document]
Comment (patorjk) [reply src/MUIDataTable.js:1508]: That's a good idea, I'll add this in too.

File: src/components/TableBody.js (hunk 2 above)
Comment (patorjk) [reply src/components/TableBody.js:249]: They will all still work. These classes are just to remove the extra line that occurs under the last cell when its stacked.

File: src/MUIDataTable.js [hunk 5]
```diff
@@ -29,19 +29,28 @@ const defaultTableStyles = theme => ({
   tableRoot: {
     outline: 'none',
   },
+  responsiveBase: {
+    overflowX: 'auto',
```
Comment (wdh2100) [inline src/MUIDataTable.js:33]: Looks like a duplicate property.

overflow, overflowX

Comment (patorjk) [reply src/MUIDataTable.js:33]: It's the base responsive class for the vertical, standard, and simple options. It takes the place of classes like responsiveScroll and responsiveScrollMaxHeight.

Comment (wdh2100) [reply src/MUIDataTable.js:33]:
```css
responsiveBase: {
    **overflowX: 'auto',**
    **overflow: 'auto',**
  },
```

Comment (patorjk) [reply src/MUIDataTable.js]: oooh, I see. Yeah, that should be a removed - good eye! Should be gone now.

**Comment by patorjk** [comment]
@wdh2100 anymore thoughts on this one? I'm thinking about merging it into the main v3 branch so I can put together a beta release.

**Comment by wdh2100** [comment]
@patorjk 
I think it's good

**Pull request merged**